presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java

```diff
@@ -317,7 +317,12 @@ public boolean supportClientTransaction()
         for (String header : splitSessionHeader(servletRequest.getHeaders(headerName))) {
             List<String> nameValue = Splitter.on('=').trimResults().splitToList(header);
             assertRequest(nameValue.size() == 2, "Invalid %s header", headerName);
-            properties.put(nameValue.get(0), nameValue.get(1));
+            try {
+                properties.put(nameValue.get(0), urlDecode(nameValue.get(1)));
+            }
+            catch (IllegalArgumentException e) {
+                throw badRequest(format("Invalid %s header: %s", headerName, e));
```

Comment:
can you please have a test for this?

Reply:
What exactly would you like to test?
Note that i added a test case in `TestHttpRequestSessionContext`

Reply:
How about adding a method
```java
properties.put(nameValue.get(0), decodeHeaderValue(nameValue.get(1), headerName));
```
That way the `catch (IllegalArgumentException)` has the smallest possible scope

Reply:
Good idea! In this case, the `com.google.common.collect.ImmutableMap.Builder#put(K, V)` cannot really throw `IllegalArgumentException`, so i will leave this as it is.